### PR TITLE
Fix Homie trigger channels (fixes #6497)

### LINKED ISSUE #6497: [mqtt-things] Homie properties with "$retained"=false attribute are not correctly handled
It's me again, the official Homie implementation tester :smile: 

I have tried to test the non-retained Homie properties (which should be modeled as Trigger channels in ESH) and they don't behave as expected. I don't see any ChannelTriggeredEvent entry in my log file when I publish a message to the corresponding topic.

Property information:
- "homie/homie-3-sensor/temperature/button/$name" -> "Thermostat Button"
- "homie/homie-3-sensor/temperature/button/$retained" -> "false"
- "homie/homie-3-sensor/temperature/button/$datatype" -> "enum"
- "homie/homie-3-sensor/temperature/button/$format" -> "SHORT_PRESSED,DOUBLE_PRESSED,LONG_PRESSED"

Message sent:
- "homie/homie-3-sensor/temperature/button" -> "SHORT_PRESSED"

But nothing appears in the log file.

IMO, this channel should be described, using XML syntax, as:
```xml
<channel-type id="trigger-channel">
    <kind>trigger</kind>
    <label>Thermostat Button</label>
    <event>
        <options>
            <option value="SHORT_PRESSED">pressed</option>
            <option value="DOUBLE_PRESSED">double pressed</option>
            <option value="LONG_PRESSED">long pressed</option>
        </options>
    </event>
</channel-type>
```
[NOTE: Although the ideal solution would be to model it as a system.button trigger channel (more info [here](https://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html#system-trigger-channel-types)).]

Anyway, the main goal of my message is to point that the channel is not behaving as a trigger channel. The representation of the channel in the Paper UI is not correct, here you have an imagen of how it looks like:

homie-event-attribute

Comparing the channel to the trigger channels in the astro binding, it shouldn't have an item-type field (the 'String' at the bottom of the channelUID), because it can't be linked to an item (well, that's not entirely correct, they can be linked to an item with a profile, but this is not supported in the binding).

Also, having a look at the source code (the part related to the Homie implementation), I haven't found any "**triggerChannel**" call, which is the function used to fire an event in a trigger channel. All the channel updates are handled by the AbstractMQTTThingHandler class, but that class treats all channel updates as state updates with the updateState function. This is, AFAIK, the main problem: **trigger channels are not correctly updated using the triggerChannel function**.

What do you think @davidgraeff? Is the triggerChannel function used in the Homie code? Is there anything I haven't taken into account?

Your work in the Mqtt binding is amazing, thanks you sooo much!

Best regards,

Aitor

### PR DESCRIPTION
**Pull request by bodiroga**
This PR fixes the problem described in the #6497 issue.

Signed-off-by: Aitor Iturrioz <riturrioz@gmail.com>

### CODE REVIEW

**Comment by bodiroga** [comment]
I have signed the ECA, but the request is in the processing state.

Looking forward to my first contribution!

**Comment by bodiroga** [comment]
This is what I can read when I try to check if my email address is verified (using the ECA Validation Tool):

> There is a valid ECA on file for riturrioz@gmail.com but we found a problem. This user is missing from the ECA LDAP group. If the problem persists after 10 minutes, please contact webmaster@eclipse.org. (eca-missing-ldap-05)

Is there anything I need to do or it is just a matter a time until everything is approved?

**Comment by htreu** [comment]
I did also check the ECA, its valid.
@bodiroga please contact the eclipse.org webmaster with the error you got. They will take care.

@davidgraeff please have a look here, thanks.

File: extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/generic/ChannelState.java [hunk 1]
```diff
@@ -77,7 +77,7 @@ public ChannelState(ChannelConfig config, ChannelUID channelUID, Value value,
         this.channelUID = channelUID;
         this.value = value;
         this.readOnly = StringUtils.isBlank(config.commandTopic);
-        this.trigger = StringUtils.isBlank(config.stateTopic);
+        this.trigger = config.trigger;
```
Comment (davidgraeff) [inline extensions/binding/org.eclipse.smarthome.binding.mqtt.generic/src/main/java/org/eclipse/smarthome/binding/mqtt/generic/internal/generic/ChannelState.java]: Please remove the `trigger` field completely and use `config.trigger`

**Pull request merged**